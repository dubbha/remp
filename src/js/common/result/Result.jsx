import React from 'react';
import PropTypes from 'prop-types';
import SearchResult from './SearchResult';
import FilmResult from './FilmResult';
import ErrorBoundary from '../errorBoundary';
import filmPropShape from '../utils/propShapes';
import './style.sass';

const Result = ({
  results,
  film,
  sortBy,
  onSortByChange,
  sortByParams,
}) => (
  <div className="result">
    <ErrorBoundary>
      {
        film
          ? (
            <FilmResult film={film} />
          )
          : (
            <SearchResult
              results={results}
              sortBy={sortBy}
              onSortByChange={onSortByChange}
              sortByParams={sortByParams}
            />
          )
      }
    </ErrorBoundary>
  </div>
);

Result.defaultProps = {
  results: [],
  film: null,
  sortBy: null,
  onSortByChange: null,
  sortByParams: null,
};

Result.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape(filmPropShape)),
  film: PropTypes.shape(filmPropShape),
  sortBy: PropTypes.string,
  onSortByChange: PropTypes.func,
  sortByParams: PropTypes.arrayOf(PropTypes.string),
};

export default Result;
