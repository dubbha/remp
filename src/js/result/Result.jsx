import React from 'react';
import PropTypes from 'prop-types';
import SearchResult from './SearchResult';
import FilmResult from './FilmResult';
import './style.sass';

const Result = ({ results, film }) => (
  <div className="result">
    {
      film
        ? <FilmResult film={film} />
        : <SearchResult results={results} />
    }
  </div>
);

Result.defaultProps = {
  results: [],
  film: {},
};

const filmPropShape = PropTypes.shape({
  poster: PropTypes.string,
  title: PropTypes.string,
  rating: PropTypes.string,
  category: PropTypes.string,
  release_year: PropTypes.string,
  runtime: PropTypes.string,
  summary: PropTypes.string,
  director: PropTypes.string,
  show_cast: PropTypes.string,
  show_id: PropTypes.number,
});

Result.propTypes = {
  results: PropTypes.arrayOf(filmPropShape),
  film: filmPropShape,
};

export default Result;
