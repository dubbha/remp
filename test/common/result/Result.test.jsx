import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Result from 'common/result/Result';

describe('FilmResult', () => {
  let props;

  beforeEach(() => {
    props = {
      film: { id: 1 },
      results: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }],
      sortBy: 'rating',
      sortByParams: ['release date', 'rating'],
      onSortByChange: jest.fn(),
    };
  });

  it('should render film result if film is available', () => {
    const wrapper = shallow(<Result {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render search result otherwise', () => {
    props.film = null;
    const wrapper = shallow(<Result {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
