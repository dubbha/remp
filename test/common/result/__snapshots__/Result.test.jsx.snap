// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmResult should render film result if film is available 1`] = `
<div
  className="result"
>
  <ErrorBoundary>
    <Result
      film={
        Object {
          "id": 1,
        }
      }
    />
  </ErrorBoundary>
</div>
`;

exports[`FilmResult should render search result otherwise 1`] = `
<div
  className="result"
>
  <ErrorBoundary>
    <SearchResult
      onSortByChange={[Function]}
      results={
        Array [
          Object {
            "id": 1,
          },
          Object {
            "id": 2,
          },
          Object {
            "id": 3,
          },
          Object {
            "id": 4,
          },
          Object {
            "id": 5,
          },
        ]
      }
      sortBy="rating"
      sortByParams={
        Array [
          "release date",
          "rating",
        ]
      }
    />
  </ErrorBoundary>
</div>
`;
